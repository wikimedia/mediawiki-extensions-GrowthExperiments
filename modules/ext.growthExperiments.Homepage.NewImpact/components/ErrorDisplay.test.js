jest.mock( '../../vue-components/icons.json', () => ( {
	cdxIconEdit: '',
	cdxIconUserTalk: '',
	cdxIconClock: '',
	cdxIconChart: '',
	cdxIconInfo: '',
	cdxIconInfoFilled: ''
} ), { virtual: true } );
const { mount } = require( '@vue/test-utils' );
const ErrorDisplay = require( './ErrorDisplay.vue' );

const renderComponent = ( props, provide ) => mount( ErrorDisplay, {
	props,
	components: {
		CText: require( '../../vue-components/CText.vue' ),
		CScoreCards: jest.fn( () => 'ScoreCardsMock' )
	},
	global: {
		provide: Object.assign( {
			RENDER_MODE: 'desktop'
		}, provide )
	}
} );

describe( 'ErrorDisplay', () => {
	const INPUT_DATA = [
		{
			provide: {
				RELEVANT_USER_USERNAME: 'Alice',
				RENDER_IN_THIRD_PERSON: false
			}
		},
		{
			provide: {
				RELEVANT_USER_USERNAME: 'Alice',
				RENDER_IN_THIRD_PERSON: true
			}
		}
	];
	for ( const input of INPUT_DATA ) {
		it( `displays appropriate text when disabled ( third person: ${ input.provide.RENDER_IN_THIRD_PERSON })`, () => {
			const wrapper = renderComponent( {}, input.provide );
			expect( wrapper.element ).toMatchSnapshot();
		} );
	}
} );
