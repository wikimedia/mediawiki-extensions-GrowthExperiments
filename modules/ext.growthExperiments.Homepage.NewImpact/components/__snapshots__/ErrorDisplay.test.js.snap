// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay displays appropriate text when disabled ( third person: false) 1`] = `
<div
  class="ext-growthExperiments-ErrorDisplay"
>
  ScoreCardsMock
  <div
    class="ext-growthExperiments-ErrorDisplay__display"
  >
    <h3
      class="ext-growthExperiments-CText ext-growthExperiments-CText--weight-bold"
    >
      
      growthexperiments-homepage-impact-error-data-not-found-text
      
    </h3>
    <div
      class="ext-growthExperiments-ErrorDisplay__image"
    />
    <div
      class="ext-growthExperiments-CText ext-growthExperiments-CText--color-subtle ext-growthExperiments-ErrorDisplay__subtext"
    >
      
      growthexperiments-homepage-impact-error-data-not-found-subtext
      
    </div>
  </div>
</div>
`;

exports[`ErrorDisplay displays appropriate text when disabled ( third person: true) 1`] = `
<div
  class="ext-growthExperiments-ErrorDisplay"
>
  ScoreCardsMock
  <div
    class="ext-growthExperiments-ErrorDisplay__display"
  >
    <h3
      class="ext-growthExperiments-CText ext-growthExperiments-CText--weight-bold"
    >
      
      growthexperiments-homepage-impact-error-data-not-found-text-third-person
      
    </h3>
    <div
      class="ext-growthExperiments-ErrorDisplay__image"
    />
    <div
      class="ext-growthExperiments-CText ext-growthExperiments-CText--color-subtle ext-growthExperiments-ErrorDisplay__subtext"
    >
      
      growthexperiments-homepage-impact-error-data-not-found-subtext-third-person
      
    </div>
  </div>
</div>
`;
